algo/uidlist.go

```diff
@@ -233,7 +238,8 @@ func IntersectWithJump(u, v []uint64, o *[]uint64) (int, int) {
 // IntersectWithBin is based on the paper
 // "Fast Intersection Algorithms for Sorted Sequences"
 // https://link.springer.com/chapter/10.1007/978-3-642-12476-1_3
-func IntersectWithBin(d, q []uint64, o *[]uint64) {
+// Returns where to move the second array(q) to. O means not found
+func IntersectWithBin(d, q []uint64, o *[]uint64) int {
```

Comment:
This is really weird function now, it puts half of the return value in the input and returns the other half.

Reply:
All of our functions do the same, checkout IntersectWithJump, IntersectWithLin